### PR TITLE
Libwebsockets: add and use 4.2.0 as default

### PR DESCRIPTION
**Pull request by Mindavi**
###### Motivation for this change

Add a libwebsockets version 4.2.0 and make it default. I think it's most logical to track the latest version, and I got everything to build.

Depends on https://github.com/NixOS/nixpkgs/pull/123135 for a successful ttyd build.

Please also look at the commit messages, I'm not sure if they're fully conforming the standards.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
    - [ ] ardour: already broken on hydra
    - [x] ttyd: bumping version makes it compile (https://github.com/NixOS/nixpkgs/pull/123135)
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6502,8 +6502,9 @@ in
     libwebsockets_3_1
     libwebsockets_3_2
     libwebsockets_4_0
```
Comment (fabaff) [inline pkgs/top-level/all-packages.nix:6504]: Is not used anywhere. Looks save to remove.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -6502,8 +6502,9 @@ in
...
-    libwebsockets_4_1;
-  libwebsockets = libwebsockets_3_2;
+    libwebsockets_4_1
```
Comment (fabaff) [inline pkgs/top-level/all-packages.nix:6505]: Also, not used.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123137` at 7d0eb78b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/fbf9fa4259521bc8f337ce8c613d89c7">ttyd</a></li>
  </ul>
</details>
<details>
  <summary>3 packages skipped due to time constraints:</summary>
  <ul>
    <li>ardour</li>
    <li>collectd</li>
    <li>collectd-data</li>
  </ul>
</details>
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>domoticz</li>
    <li>driftnet</li>
    <li>libwebsockets (libwebsockets_4_2)</li>
    <li>libwebsockets_3_1</li>
    <li>libwebsockets_3_2</li>
    <li>libwebsockets_4_0</li>
    <li>libwebsockets_4_1</li>
    <li>mosquitto</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by fabaff — COMMENTED** [review]
Mosquitto seems the only one that requires the 3.x serie but that's something for another PR.

**Comment by Mindavi** [comment]
I was already wondering if I should keep around all those old versions. Thanks for pushing me over the edge to do so :). Mosquitto compiles fine with libwebsockets 4.2, so we can bump that one / move to the default version and then remove all the 3.x series, I think. Will prepare a PR for that.

**Comment by Mindavi** [comment]
@fabaff got some time to look at this? Should be good to go.

**Pull request merged**